### PR TITLE
Fix reported CVEs

### PR DESCRIPTION
**Pull request by tejaswini-imply**
- Suppress CVEs from dependencies with no available fix or false positives
  - `hadoop-annotations`: CVE-2022-25168, CVE-2021-33036
  - `hadoop-client-runtime`: CVE-2023-1370, CVE-2023-37475
  - `okio`: CVE-2023-3635
- Upgrade grpc version to fix CVE-2023-33953

### CODE REVIEW

File: owasp-dependency-check-suppressions.xml [hunk 1]
```diff
@@ -359,6 +359,9 @@
     ]]></notes>
     <cve>CVE-2022-45855</cve>
     <cve>CVE-2022-42009</cve>
+    <!-- Suppress hadoop CVEs that not applicable to hadoop-annotations -->
```
Comment (abhishekagarwal87) [inline owasp-dependency-check-suppressions.xml:362]: we should add a comment as to why these are not applicable to hadoop-annotations.

Comment (tejaswini-imply) [reply owasp-dependency-check-suppressions.xml:362]: Updated it! Thanks, @abhishekagarwal87

File: distribution/pom.xml [hunk 2]
```diff
@@ -448,8 +448,6 @@
                                         <argument>org.apache.druid.extensions.contrib:aliyun-oss-extensions</argument>
                                         <argument>-c</argument>
                                         <argument>org.apache.druid.extensions.contrib:opentelemetry-emitter</argument>
-                                        <argument>-c</argument>
-                                        <argument>org.apache.druid.extensions:druid-iceberg-extensions</argument>
```
Comment (abhishekagarwal87) [inline distribution/pom.xml:452]: why are you removing this? This profile is bundle-contrib-exts which is supposed to package all contrib extensions in the bundle.

Comment (tejaswini-imply) [reply distribution/pom.xml]: Reverted this! Thanks for pointing it out.

**Review by abhishekagarwal87 — APPROVED** [review]

**Pull request merged**